c7n/resources/rdscluster.py

```diff
@@ -244,15 +243,23 @@ def process_snapshot_retention(self, client, cluster):
             self.set_retention_window(client, cluster, new_retention)
 
     def set_retention_window(self, client, cluster, retention):
-        _run_cluster_method(
-            client.modify_db_cluster,
-            dict(DBClusterIdentifier=cluster['DBClusterIdentifier'],
-                 BackupRetentionPeriod=retention,
-                 PreferredBackupWindow=cluster['PreferredBackupWindow'],
-                 PreferredMaintenanceWindow=cluster['PreferredMaintenanceWindow']),
-            (client.exceptions.DBClusterNotFoundFault, client.exceptions.ResourceNotFoundFault),
-            client.exceptions.InvalidDBClusterStateFault)
-
+        engine_mode = cluster.get('EngineMode', 0)
+        if engine_mode == 'serverless':
+            _run_cluster_method(
+                client.modify_db_cluster,
+                dict(DBClusterIdentifier=cluster['DBClusterIdentifier'],
+                    BackupRetentionPeriod=retention),
+                (client.exceptions.DBClusterNotFoundFault, client.exceptions.ResourceNotFoundFault),
+                client.exceptions.InvalidDBClusterStateFault)
+        else:
+            _run_cluster_method(
+                client.modify_db_cluster,
+                dict(DBClusterIdentifier=cluster['DBClusterIdentifier'],
+                    BackupRetentionPeriod=retention,
+                    PreferredBackupWindow=cluster['PreferredBackupWindow'],
```

Comment:
This action is for setting log retention. I was wondering if we even needed reapplying `PreferredBackupWindow` and `PreferredMaintenanceWindow` at all re: modifying log retention. I tested it with serverless and aurora-mysql, modifying `BackupRetentionPeriod` works without reapplying the other two configuration. However, I might have to do more investigation wrt other engine types

Reply:
So this is specifically for Serverless v1 which is kind of deprecated. However you can still create it using this [link](https://catalog.us-east-1.prod.workshops.aws/workshops/098605dc-8eee-4e84-85e9-c5c6c9e43de2/en-US/lab10-aurora-serverless/setup).

[Here](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-rds-dbcluster.html) is the limitation on v1. 

_Currently, when you are updating the stack for an Aurora Serverless DB cluster, you can't include changes to any other properties when you specify one of the following properties: PreferredBackupWindow, PreferredMaintenanceWindow, and Port. This limitation doesn't apply to provisioned DB clusters._

Reply:
yup, i got that. My comment was more from an implementation pov. Currently, we have:
```
        _run_cluster_method(
            client.modify_db_cluster,
            dict(DBClusterIdentifier=cluster['DBClusterIdentifier'],
                 BackupRetentionPeriod=retention,
                 PreferredBackupWindow=cluster['PreferredBackupWindow'],
                 PreferredMaintenanceWindow=cluster['PreferredMaintenanceWindow']),
```
I was thinking whether the 2 parameters aka `PreferredBackupWindow` and `PreferredMaintenanceWindow` were required as part of this action?
Could we just do?
```
        _run_cluster_method(
            client.modify_db_cluster,
            dict(DBClusterIdentifier=cluster['DBClusterIdentifier'],
                 BackupRetentionPeriod=retention),
```